target/cheri-common/op_helper_cheri_common.c

```diff
@@ -236,18 +241,24 @@ void CHERI_HELPER_IMPL(cjalr(CPUArchState *env, uint32_t cd, uint32_t cb,
     if (cap_is_sealed_entry(cbp)) {
         // If we are calling a "sentry" cap, remove the sealed flag
         cap_unseal_entry(&next_pcc);
+        assert(cap_get_cursor(&next_pcc) == addr &&
+               "Should have raised an exception");
+    } else {
+        // Can never create an unrepresentable capability since we
+        // bounds-checked the jump target.
+        assert(is_representable_cap_with_addr(&next_pcc, addr) &&
```

Comment:
Maybe make this a `cheri_debug_assert` to avoid checking it twice?

Reply:
Although I guess this will be a simple in-bounds check that the compiler might be able to elide if it inlines `validate_jump_target`